### PR TITLE
cisco_secure_email_gateway: update ingest pipeline to handle v15

### LINKED ISSUE #7806: cisco_secure_email_gateway: update pipeline to handle changes in v15
Cisco release note : https://www.cisco.com/c/dam/en/us/td/docs/security/esa/esa15-0/release_notes/Secure_Email_15-0_Release_Notes.pdf

From this release onwards, the following Common Event Format (CEF) field names are changed for the Consolidated Event logs:

- `endTime` to `end`
- `startTime` to `start`
- `sourceAddress` to `src`
- `sourceHostName` to `shost`

### LINKED ISSUE #7877: [cisco_secure_email_gateway] Missing event.start mapping
It appears that a mapping is missing for `event.start` in the cisco_secure_email_gateway integration.

Screenshot 2023-09-19 at 14 35 55

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

The release [notes](https://www.cisco.com/c/dam/en/us/td/docs/security/esa/esa15-0/release_notes/Secure_Email_15-0_Release_Notes.pdf) for v15 note that some fields have changed name:

- `endTime` => `end`
- `startTime` => `start`
- `sourceAddress` => `src`
- `sourceHostName` => `shost`

To maintain backwards compatibility, provide alternations for both cases as pattern definitions in the relevant grok.

After this change, the only place that the old names are found in the elasticsearch directory is in the pattern definitions.

```
$ rg '(endTime|startTime|sourceAddress|sourceHostName)' data_stream/log/elasticsearch
data_stream/log/elasticsearch/ingest_pipeline/pipeline_consolidated_event.yml
58:        STARTTIME: '(startTime|start)'
59:        ENDTIME: '(endTime|end)'
60:        SOURCEADDRESS: '(sourceAddress|src)'
61:        SOURCEHOSTNAME: '(sourceHostName|shost)'
```

~Ideally, we would have test cases for this, but they are not available at this stage.~ This is now added.

The change also fixes the handling of logs that do not include a syslog priority, and the conversion of `event.{start,end}` to correctly formatted dates, and their type mappings.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Closes #7806
- Closes #7877

### CODE REVIEW

**Review by bhapas — COMMENTED** [review]
There seem to be some examples here 
https://www.cisco.com/c/en/us/td/docs/security/ces/ces_15-0/user_guide/b_ESA_Admin_Guide_ces_15-0/b_ESA_Admin_Guide_12_1_chapter_0100111.html

Can we not use them in pipeline tests additional to the existing ones?

**Comment by efd6** [comment]
@bhapas I've taken a look at those examples and none exercises the changes here. Adding them as tests would probably be a sensible thing to do in the long term, but it would be sensible to have a directed plan of attack. @LaZyDK Do you have suggestions about a priority ordering of messages in that example set?

**Comment by bhapas** [comment]
@efd6 Yes. Just that we release a package without any tests on it. That was my only concern.

**Comment by efd6** [comment]
It's not that it has no tests, it just has no tests for this change. Taking the examples in that document, according to the release notes, doesn't appear to change that. There are a lot of examples in the document, and they are not structured in a way that makes it obvious which to include.

**Comment by bhapas** [comment]
@efd6 If it the `Consolidated Event Logs` Format that is changed

Then we have an example is it?

```
Example of Consolidated Event Logs

In this example, the log shows all the available fields selected when you configure a log subscription with the log type as Consolidated Event Logs.

Thu Jun 30 08:04:50 2022: CEF:0|Cisco|C100V Email Security Virtual Appliance|14.0.0-657
|ESA_CONSOLIDATED_LOG_EVENT|Consolidated Log Event|5|deviceExternalId=42127C7DDEE76852677B-F80CE8074CD3
ESACustomLogs={'label2': ['value20'], 'label1': ['value1', 'value2']} 
ESALogHeaders={'reply-to': ['test@esa.com ', 'newsletter@esa.com'], 'from': ['any@esa.com']}
ESAMID=1053 ESAICID=134 ESAAMPVerdict=UNKNOWN ESAASVerdict=NEGATIVE ESAAVVerdict=NEGATIVE 
ESACFVerdict=MATCH end=Thu Mar 18 08:04:46 2021 ESADLPVerdict=NOT_EVALUATED 
dvc=10.10.193.13 ESAAttachmentDetails={'test.txt': {'AMP': {'Verdict': 'FILE UNKNOWN', 
'fileHash': '7f843d263304fb0516d6210e9de4fa7f01f2f623074aab6e3ee7051f7b785cfa'}, 'BodyScanner':
{'fsize': 10059}}} ESAFriendlyFrom=test@esa.com ESAGMVerdict=NEGATIVE start=Thu Mar 18 08:04:29 2021 
deviceInboundInterface=Incomingmail deviceDirection=0 ESAMailFlowPolicy=ACCEPT suser=test@esa.com 
cs1Label=MailPolicy cs1=DEFAULT ESAMFVerdict=NOT_EVALUATED act=QUARANTINED ESAFinalActionDetails=To POLICY 
cs4Label=ExternalMsgID cs4='<20210318070601.40490.18684@mail1.example.com>' ESAMsgSize=11873 ESAOFVerdict=POSITIVE 
duser=9076@testing.com ESAHeloIP=10.11.1.2 cfp1Label=SBRSScore cfp1=None ESASDRDomainAge=27 years 2 months 15 days
cs3Label=SDRThreatCategory cs3=N/A cs6Label=SDRRepScore cs6=Weak ESASPFVerdict={'mailfrom': {'result': 'None', 
'sender': 'test@esa.com'}, 'helo': {'result': 'None', 'sender': 'postmaster'}, 'pra': {'result': 'None', 'sender':
'test@esa.com'}}shost=unknown ESASenderGroup=UNKNOWNLIST src=10.11.1.2 msg='Testing'
```

**Comment by LaZyDK** [comment]
@efd6 I would add the above Consolidated Event log to the tests. I think that it coveres what you need.

**Comment by efd6** [comment]
Yeah, we had a discussion about this and agreed that this is probably what needs to be done. The log line there fails to parse out the details, but I am not convinced of copy/pasted text from documentation, so ideally we could find examples from real instances.

File: packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-consolidated-event.log-expected.json [hunk 1]
```diff
@@ -1514,6 +1514,118 @@
...
...23 deviceInboundInterface=IncomingMail deviceDirection=0 ESAMailFlowPolicy=ACCEPT suser=0101016d24913de4eb-28ea32e6-4dad-4943-b6f7-ef0dddbb875d-000000@bounce.com cs1Label=MailPolicy cs1=Exception_support cs2Label=SenderCountry cs2=United States ESAMFVerdict=MATCH act=DELIVERED cs4Label=ExternalMsgID cs4='' ESAMsgSize=3518 ESAOFVerdict=NOT_EVALUATED duser=support@example.com ESAHeloDomain=smtp-out.us-west-1.amazon.com ESAHeloIP=216.160.83.56 cfp1Label=SBRSScore cfp1=3.5 ESASDRDomainAge=30 days (or greater) cs3Label=SDRThreatCategory cs3=N/A cs6Label=SDRRepScore cs6=Favorable ESASPFVerdict={'mailfrom': {'result': 'Pass', 'sender': '0101016d24913de4eb-28ea32e6-4dad-4943-b6f7-ef0dddbb875d-000000@bounce.com'}, 'helo': {'result': 'None', 'sender': 'postmaster@smtp-out.us-west-1.amazon.com'}} shost=smtp-out.us-west-1.amazon.com ESASenderGroup=ACCEPTLIST src=216.160.83.56 msg='Alert - wireless - APs came up' ESATLSInCipher=ECDHE-RSA-AES256-GCM-SHA384 ESATLSInConnStatus=Success ESATLSInProtocol=TLSv1.2 ESATLSOutCipher=ECDHE-RSA-AES256-GCM-SHA384 ESATLSOutConnStatus=Success ESATLSOutProtocol=TLSv1.2 ESAAMPVerdict=SKIPPED ESAASVerdict=NEGATIVE ESAAVVerdict=NEGATIVE ESACFVerdict=NOT_EVALUATED"
```
Comment (bhapas) [inline packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-consolidated-event.log-expected.json:1585]: Should this field be present?

Comment (efd6) [reply packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-consolidated-event.log-expected.json:1585]: It's in [field.yml](https://github.com/elastic/integrations/blob/d024837448065f425df01c21b0e8d0fac30219f8/packages/cisco_secure_email_gateway/data_stream/log/fields/fields.yml#L352-L353), so yes.

Comment (bhapas) [reply packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-consolidated-event.log-expected.json:1585]: But isn't this same as `event.original` or am I mixing up?

Comment (efd6) [reply packages/cisco_secure_email_gateway/data_stream/log/_dev/test/pipeline/test-common-consolidated-event.log-expected.json:1585]: No, you're right it's not. It's the message of the syslog entry.

**Review by bhapas — APPROVED** [review]
LGTM

**Pull request merged**